site-content/source/modules/ROOT/pages/blog/Apache-Cassandra-Changelog-11-January-2022.adoc

```diff
@@ -0,0 +1,114 @@
+
+=== Added
+
+An https://cwiki.apache.org/confluence/display/CASSANDRA/Roadmap[up-to-date roadmap^] for the project has been added to the wiki.
+
+=== Added
+
+Andrés de la Peña opened a discussion about https://cwiki.apache.org/confluence/display/CASSANDRA/CEP-3%3A+Guardrails[CEP-3^], a guardrails feature that is intended to be an easy way to “enforce system-wide soft and hard limits to prevent anti-patterns of bad usage”. In the long-term this is intended to prevent the severe degradation of a node/cluster’s performance through user actions such as having too many secondary indexes, too large partitions, almost full disks, and so on. This feature passed a vote, has been developed and merged! Congrats Andrés!
+
+=== Added
+
+https://cwiki.apache.org/confluence/display/CASSANDRA/CEP-10%3A+Cluster+and+Code+Simulations[CEP-10 Cluster and Code Simulation^] passed a vote and Benedict Elliot Smith has merged in the simulator changes so can expect to see some interesting things showing up regarding correctness on our system from the new testing paradigms this enables.
+
+=== Discussed
+
+Josh McKenzie is seeking more feedback on the topic of keeping a releasable trunk and our quality on the project. You can join https://lists.apache.org/thread/8xt4tqb3w4j3jyxj859o3vs8f5xjgqg8[the discussion here^]
+
+=== Passed
+
+After clarification that the API for distributed transactions will be modular / pluggable, CEP-15 General Purpose Transactions passed. This work will allow for experimentation in the future with other algorithms. As a community we're excited to see how the project evolves!
+
+=== Passed
+
+
+https://cwiki.apache.org/confluence/display/CASSANDRA/CEP-16%3A+Auth+Plugin+Support+for+CQLSH[CEP-16 Auth Plugin Support for CQLSH^], proposed by Stefan Miklosovic, will enable switching to a new enhanced authentication scheme while retaining the ability to use cqlsh in normal operation.  A vote was taken and passed and the feature is in development.
+
+=== Passed
+
+https://cwiki.apache.org/confluence/display/CASSANDRA/CEP-17%3A+SSTable+format+API[CEP-17^] by Jacek Lewandowski is around an SSTable format API. The proposal is for a design that enables for optionally plugging-in alternative implementations of SSTables and for testing the individual implementations in isolation. You can follow the https://lists.apache.org/thread.html/r636bebcab4e678dbee042285449193e8e75d3753200a1b404fcc7196%40%3Cdev.cassandra.apache.org%3E[discussion here^]. A vote was taken and passed and this CEP is in development.
+
+=== Closed
+
+CEP-18 was under discussion to add more modularity to a few different subsystems. It was decided to close the CEP to enable ideas to be considered individually, rather than grouped under one feature proposal.
```

Comment:
Add link to CEP-18:

```suggestion
https://cwiki.apache.org/confluence/x/5YtnCw[CEP-18^] was under discussion to add more modularity to a few different subsystems. It was decided to close the CEP to enable ideas to be considered individually, rather than grouped under one feature proposal.
```